src/main/java/org/springframework/data/repository/config/CustomRepositoryImplementationDetector.java

```diff
@@ -47,6 +50,7 @@
 	private final MetadataReaderFactory metadataReaderFactory;
 	private final Environment environment;
 	private final ResourceLoader resourceLoader;
+	private final RepositoryBeanNameGenerator nameGenerator;
```

Comment:
I wonder if there's a way that we can avoid holding an additional generator to the one that the configuration uses. I could live with the CDI interaction becoming a bit more complicated for now as we're going to unify that into a dedicated configuration interface anyway and it would even make the requirements for such a unified interface even more explicit.

So let's say the `detectCustomImplementation(…)` overload taking a lot of parameters doesn't have any bounds for now. Let's make that as complicated as necessary to then know what needs to go into the shared config interface.